astropy/io/misc/tests/test_hdf5.py

```diff
@@ -545,3 +565,205 @@ def test_read_h5py_objects(tmpdir):
     assert np.all(t4['a'] == [1, 2, 3])
 
     f.close()         # don't raise an error in 'test --open-files'
+
+
+def assert_objects_equal(obj1, obj2, attrs, compare_class=True):
+    if compare_class:
+        assert obj1.__class__ is obj2.__class__
+
+    info_attrs = ['info.name', 'info.format', 'info.unit', 'info.description', 'info.meta']
+    for attr in attrs + info_attrs:
+        a1 = obj1
+        a2 = obj2
+        for subattr in attr.split('.'):
+            try:
+                a1 = getattr(a1, subattr)
+                a2 = getattr(a2, subattr)
+            except AttributeError:
+                a1 = a1[subattr]
+                a2 = a2[subattr]
+
+        # Mixin info.meta can None instead of empty OrderedDict(), #6720 would
+        # fix this.
+        if attr == 'info.meta':
+            if a1 is None:
+                a1 = {}
+            if a2 is None:
+                a2 = {}
+
+        assert np.all(a1 == a2)
+
+# Testing FITS table read/write with mixins.  This is mostly
+# copied from ECSV mixin testing.
```

Comment:
It does make one wonder whether this should be unified...  but another time is fine.

Reply:
I tend to apply a lower bar for testing code, but indeed I had the same thought.